### PR TITLE
Feat: lesson22

### PR DESCRIPTION
**Pull request by yuka830**
# [Lesson22](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#22)
```
同じことを年齢でもやってください。
```
## [CodeSandbox](https://codesandbox.io/s/lesson22-8j3ub?file=/main.js)
Last commit: [314f667](https://github.com/yuka830/js-lessons/pull/28/commits/314f667928f0cf411c76740ba831a5699161c92c)

- Sort buttons
 Created `targetSortElements` array which includes elements for sorting, and made sort arrow buttons with it in the looping.
 If we want to sort other element, just add their id name to the array(number content only now)
- Sorting function
 Got button elements with `querySelectorAll`, and added click event to each of them in the looping.
 Got parent element's id name of each button with `parentNode.id`, and used them in each sort function.
 Implemented new function `initSortAndOthersSortImg`, set to start with ascending order every time and init others sort arrow button image.

### CODE REVIEW

File: lesson22/main.js [hunk 1]
```diff
@@ -145,56 +145,85 @@ const createSortBtn = () => {
...
 };
 
 const clickSortBtn = (usersData) => {
-  const sortArrow = document.querySelector(".sort-btn");
-  const sortArrowImg = document.querySelector(".sort-img");
-  sortArrow.addEventListener("click", () => {
-    changeSortStateAndArrowImg(sortArrowImg);
-    rerenderTableData(usersData);
+  const sortArrowBtns = document.querySelectorAll(".sort-btn");
+  Array.from(sortArrowBtns).forEach((sortArrowBtn) => {
```
Comment (kenmori) [inline lesson22/main.js]:
```suggestion
sortArrowBtns.forEach((sortArrowBtn) => {
```
これでもでも動き一緒になりますか?

File: lesson22/main.js [hunk 2]
```diff
@@ -145,56 +145,85 @@ const createSortBtn = () => {
...
+  });
 };
 
 const clickSortBtn = (usersData) => {
-  const sortArrow = document.querySelector(".sort-btn");
-  const sortArrowImg = document.querySelector(".sort-img");
-  sortArrow.addEventListener("click", () => {
-    changeSortStateAndArrowImg(sortArrowImg);
-    rerenderTableData(usersData);
+  const sortArrowBtns = document.querySelectorAll(".sort-btn");
```
Comment (kenmori) [inline lesson22/main.js]:
```suggestion
  const sortArrowBtns = document.querySelectorAll(".js-sort-btn");
```

File: lesson22/main.js [hunk 3]
```diff
@@ -145,56 +145,85 @@ const createSortBtn = () => {
...
+  Array.from(sortArrowBtns).forEach((sortArrowBtn) => {
+    const tHeaderId = sortArrowBtn.parentNode.id;
+    const sortArrowImg = sortArrowBtn.firstElementChild;
+    sortArrowBtn.addEventListener("click", () => {
```
Comment (kenmori) [inline lesson22/main.js]: use event object.  instead of `sortArrowBtn.parentNode.id;`
event.currentTarget.parentNode.id

File: lesson22/main.js [hunk 4]
```diff
@@ -145,56 +145,85 @@ const createSortBtn = () => {
...
+      initSortAndOthersSortImg(tHeaderId);
+      changeSortStateAndArrowImg(sortArrowImg);
```
Comment (kenmori) [inline lesson22/main.js:173]: also sortArrowImg, if you can get it from event object

File: lesson22/main.js [hunk 5]
```diff
@@ -145,56 +145,85 @@ const createSortBtn = () => {
...
   }
   if (sortState === "asc") {
     sortState = "desc";
-    sortArrow.src = "/img/desc.svg";
+    sortArrowImg.src = "/img/desc.svg";
     return;
   }
   sortState = "both";
-  sortArrow.src = "/img/both.svg";
+  sortArrowImg.src = "/img/both.svg";
```
Comment (kenmori) [inline lesson22/main.js]: IMO
i often use switch because of readable

File: lesson22/main.js [hunk 6]
```diff
@@ -145,56 +145,85 @@ const createSortBtn = () => {
...
+      const element = document.getElementById(key);
+      element.dataset.sort = "sort";
+    }
+  });
+};
+
+const renderSortBtn = (usersData) => {
+  setDataSort(usersData);
+  const elements = document.querySelectorAll("[data-sort]");
+  Array.from(elements).forEach((element) => {
```
Comment (kenmori) [inline lesson22/main.js]:
```suggestion
  elements.forEach((element) => {
```

File: lesson22/main.js [hunk 7]
```diff
@@ -120,7 +121,6 @@ const formingTableHeaderNameWithKey = (key) => {
  *Creating table data with value of usersData
  * @param {Array} usersData The Array of usersData
  */
-
```
Comment (csakyo) [inline lesson22/main.js:123]: I was thinking "what is this..?" so I googled about it and then I just knew about JSDOC. I did not know about it so I lerned a lot!

File: lesson22/main.js (hunk 1 above)
Comment (yuka830) [reply lesson22/main.js]: It works! I've misunderstood about looping of NodeList. Thank you.
[4f2be45](https://github.com/yuka830/js-lessons/pull/28/commits/4f2be45f21b29576235b85cfa840a204f85d3288)

File: lesson22/main.js (hunk 6 above)
Comment (yuka830) [reply lesson22/main.js]: I fixed it too.
[4f2be45](https://github.com/yuka830/js-lessons/pull/28/commits/4f2be45f21b29576235b85cfa840a204f85d3288)

File: lesson22/main.js (hunk 2 above)
Comment (yuka830) [reply lesson22/main.js]: I fixed it.
[324713d](https://github.com/yuka830/js-lessons/pull/28/commits/324713d61fe85e89052769bbe8c1130035707f4a)

File: lesson22/main.js (hunk 3 above)
Comment (yuka830) [reply lesson22/main.js]: In this part, I wasn't sure which way to implement them, but I will use the event object from now and then.
Thank you for teaching me.
[123dbde](https://github.com/yuka830/js-lessons/pull/28/commits/123dbdef3fe81a135b0c3ab298f8d47e441c0416)

File: lesson22/main.js (hunk 4 above)
Comment (yuka830) [reply lesson22/main.js:173]: [123dbde](https://github.com/yuka830/js-lessons/pull/28/commits/123dbdef3fe81a135b0c3ab298f8d47e441c0416)

File: lesson22/main.js (hunk 5 above)
Comment (yuka830) [reply lesson22/main.js]: I see, I think it's much better now! Thank you.
[4a0fd5a](https://github.com/yuka830/js-lessons/pull/28/commits/4a0fd5a13aab7e042ccbee72a96e4126178d0205)

File: lesson22/main.js (hunk 7 above)
Comment (yuka830) [reply lesson22/main.js:123]: Thank you so much for reviewing! Ms.Onozuka told me about this before.
I'm happy to share about this with you :)

**Review by kenmori — APPROVED** [review]

**Pull request merged**